cmd/trade.go

```diff
@@ -506,7 +506,19 @@ func runTradeCmd(options inputs) {
 	botConfig = convertDeprecatedBotConfigValues(l, botConfig)
 	l.Infof("Trading %s:%s for %s:%s\n", botConfig.AssetCodeA, botConfig.IssuerA, botConfig.AssetCodeB, botConfig.IssuerB)
 
-	userID := "-1" // TODO DS Properly generate and save user ID.
+	var userIDPrehash string
+	if botConfig.IsTradingSdex() {
+		userIDPrehash = botConfig.TradingAccount()
+	} else {
+		userIDPrehash = amplitudeAPIKey
```

Comment:
this should use the hash of the API key for the centralized exchange.
By using the amplitudeAPIKey, all users on centralized exchanges will have the same userID, which is not what we want here.